### PR TITLE
DEV-2791: Show user email & id in sentry

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a

#### What's this PR do?
Adds user email and id to sentry so that errors are identifiable with the user that got the error.

#### Why are we doing this? How does it help us?
This will be possibly be useful for a few different scenarios: 
- if we have an identifier for a contributor in Sentry we can use that to look in the logs/Stripe/Salesforce to get more detail about what happened
- if we have an ID of a Org Admin we can contact them about the issue to get more detail or assist them

With user id/email we can identify the above information.

#### How should this be manually tested? Please include detailed step-by-step instructions.
- Log into review app and try to break the app so that a sentry error is shown with the user's email/id
- Sentry error: https://sentry.io/organizations/news-revenue-hub/issues/3808783797/?environment=dev-2791&project=6046263&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=1h

image

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
no

#### Have automated unit tests been added? If not, why?
n/a

#### How should this change be communicated to end users?
n/a

#### Are there any smells or added technical debt to note?
no

#### Has this been documented? If so, where?
no

#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-2791

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
no

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
no

### CODE REVIEW

File: spa/src/hooks/useUser.ts [hunk 1]
```diff
@@ -44,8 +45,12 @@ function useUser(): UserHookResult {
     // When user logs in, the authentication endpoint returns user data and it gets
     // stored in localstorage. We should update the localstorage data each time user
     // is refetched so it isn't stale.
-    onSuccess: (data) => localStorage.setItem(LS_USER, JSON.stringify(data)),
+    onSuccess: (data: User) => {
+      Sentry.setUser({ email: data.email, id: data.id });
```
Comment (nrh-cklimas) [inline spa/src/hooks/useUser.ts:49]: Do we want to record the IP address too? https://docs.sentry.io/platforms/javascript/enriching-events/identify-user/#ip_address I know it's not in AC but it seems like we may as well while we are here.

Comment (GuiMend) [reply spa/src/hooks/useUser.ts:49]: Hey @nrh-cklimas , if you check sentry right now, it already captures the user's IP.

Comment (nrh-cklimas) [reply spa/src/hooks/useUser.ts:49]: I was looking at this:

> If the field is omitted, the default value is null. However, due to backwards compatibility concerns, certain platforms (in particular JavaScript) have a different default value for "{{auto}}". SDKs and other clients should not rely on this behavior and should set IP addresses or "{{auto}}" explicitly.

It's hard for me to fully understand what they mean by this, but it sounds like we shouldn't depend on them recording IP addresses by default?

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]

**Review by nrh-cklimas — APPROVED** [review]

File: spa/src/hooks/useUser.ts (hunk 1 above)
Comment (GuiMend) [inline spa/src/hooks/useUser.ts:49]: Adding ip address:
```suggestion
      Sentry.setUser({ email: data.email, id: data.id, ip_address: "{{auto}}" });
```

**Review by x110dc — APPROVED** [review]

**Pull request merged**